xcube/util/progress.py

```diff
@@ -99,6 +99,25 @@ def inc_work(self, work: float):
     def finish(self):
         self._finished = True
 
+    def to_dict(self):
+        d = dict()
+        if self._label:
+            d.update(label=self._label)
+        if self._total_work:
+            d.update(total_work=self._total_work)
+        if self._super_work:
+            d.update(super_work=self._super_work)
+        if self._super_work_ahead:
+            d.update(super_work_ahead=self._super_work_ahead)
+        if self._completed_work:
+            d.update(completed_work=self._completed_work)
+        if self._finished:
+            d.update(finished=self._finished)
+        if self._exc_info:
+            d.update(exc_info=self.exc_info_text)
+
+        return d
+
```

Comment:
It is fine to have this method, but this is not necessarily what should be send as JSON to the API. The following members are internal and there is no use for it outside the module: `super_work_ahead`, `super_work`

Add unit test that covers the new method..